src/app/handlers/blacklist.handler.ts

```diff
@@ -44,8 +44,9 @@ export class BlacklistHandler extends Handler {
       return;
     }
 
+    let deleted = false;
     this._expressions.forEach(async ({ regex, label }) => {
```

Comment:
Let’s change this forEach. It’s making this code messy. Instead, 

let patternsFound = expressions.filter()

…

if patternsFound is not empty, file report, delete message, etc.

This way, finding the pattern doesn’t leave side effects.

Reply:
thx